v20.2/create-role.md

```diff
@@ -38,13 +38,259 @@ The `CREATE ROLE` [statement](sql-statements.html) creates SQL [roles](authoriza
+The following statements are run by the `root` user that is a member of the `admin` role and has `ALL` privileges.
+{{site.data.alerts.end}}
+
+### Create a role
+
+Role names are case-insensitive; must start with a letter, number, or underscore; must contain only letters, numbers, periods, or underscores; and must be between 1 and 63 characters.
+
+~~~ sql
+root@:26257/defaultdb> CREATE ROLE no_options;
+~~~
+
+~~~ sql
+root@:26257/defaultdb> SHOW ROLES;
+~~~
+
+~~~
+ username  | options | member_of
+ ----------+---------+------------
+admin      |         | {}
+no_options | NOLOGIN | {}
+root       |         | {admin}
+(3 rows)
+~~~
+
+After creating roles, you must [grant them privileges to databases](grant.html).
+
+### Create a role that can log in to the database
+
+~~~ sql
+root@:26257/defaultdb> CREATE ROLE can_login WITH LOGIN PASSWORD '$tr0nGpassW0rD' VALID UNTIL '2021-10-10';
+~~~
+
+~~~ sql
+root@:26257/defaultdb> SHOW ROLES;
+~~~
+
+~~~
+ username  |                options                | member_of
+-----------+---------------------------------------+------------
+admin      |                                       | {}
+can_login  | VALID UNTIL=2021-10-10 00:00:00+00:00 | {}
+no_options | NOLOGIN                               | {}
+root       |                                       | {admin}
+(4 rows)
+~~~
+
+### Prevent a role from using password authentication
+
+The following statement prevents the role from using password authentication and mandates certificate-based client authentication:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE ROLE no_password WITH PASSWORD NULL;
+~~~
+
+~~~ sql
+root@:26257/defaultdb> SHOW ROLES;
+~~~
+
+~~~
+ username  |                options                | member_of
+-----------+---------------------------------------+------------
+admin      |                                       | {}
+can_login  | VALID UNTIL=2021-10-10 00:00:00+00:00 | {}
+no_options | NOLOGIN                               | {}
+no_password| NOLOGIN                               | {}  
+root       |                                       | {admin}
+(5 rows)
+~~~
+
+### Create a role that can create other roles and manage authentication methods for the new roles
+
+~~~ sql
+root@:26257/defaultdb> CREATE ROLE can_create_role WITH CREATEROLE CREATELOGIN;
+~~~
+
+~~~ sql
+root@:26257/defaultdb> SHOW ROLES;
+~~~
+
+~~~
+   username     |                options                | member_of
+----------------+---------------------------------------+------------
+admin           |                                       | {}
+can_create_role | CREATELOGIN, CREATEROLE, NOLOGIN      | {}
+can_login       | VALID UNTIL=2021-10-10 00:00:00+00:00 | {}
+no_options      | NOLOGIN                               | {}
+no_password     | NOLOGIN                               | {}
+root            |                                       | {admin}
+(6 rows)
+~~~
+
+### Create a role that can create and rename databases
```

Comment:
Might be worth adding an opening sentence that links to `CREATE DATABASE` and `RENAME DATABASE` stmts